dask/base.py

```diff
@@ -987,6 +990,37 @@ def _normalize_function(func):
             return str(func)
 
 
+def normalize_dataclass(obj):
+    # If dataclasses should not be compared, they should not be deterministically
+    # tokenized either, since the purpose of tokenization is comparision.
+    try:
+        comparable = obj.__dataclass_params__.eq
+    except AttributeError:
+        comparable = False
+
+    if comparable:
+        fields = [
+            (field.name, getattr(obj, field.name))
+            for field in dataclasses.fields(obj)
+            if field.compare
+        ]
+        if fields:
+            return (
+                normalize_function(type(obj)),
```

Comment:
Do we need to tokenize the actual class here, or would the name of the dataclass type be sufficient? I'm genuinely curious because we use `type(...).__name__` elsewhere

Reply:
1. Different libraries could define totally different dataclasses with the same name
2. You could change the definition of a dataclass within an interactive session, but keep the same name

It seems to me that we use `type(...).__name__` in other places where the type has already been restricted to something canonical (a list or tuple, or a NumPy matrix, etc.). (I'd argue the list/tuple case is actually insufficient, because it could be a user subclass of list/tuple, which probably shouldn't tokenize the same as a pure list/tuple.)